### ISSUE TITLE
Galil: Bring up to date

### ISSUE THREAD

**Issue by DominicOram** [ISISComputingGroup/IBEX#6255]
As a developer I would like to be on the latest version of the Galil driver. In meeting between @FreddieAkeroyd and myself we decided that to stop this becoming such an issue in the future as part of this ticket we should:
* Fork our galil repo from the "vendor" one
* Cherry-pick our current changes onto new feature branches with the aim to get these merged into the vendor code (aiming to keep history)
* Going forward we will point our build at the master of this fork, which will be the same as the upstream master. If we diverge we can temporarily point our build at a different branch but we should aim to keep the two masters in step as much as possible

## Acceptance Criteria
* The above is done for the galil repo
* There is a branch with all the amalgamated changes on, which we can test on hardware

**Comment by DominicOram** [comment]
Spun off from https://github.com/ISISComputingGroup/IBEX/issues/2830

**Comment by DominicOram** [comment]
Pulled into sprint as it's effectively just cutting #2830 in half.

**Comment by DominicOram** [comment]
I'm struggling with this ticket, mostly due to my mental health. I'm going to move on to something else but it should be in a state other people can pick it up in. If no one else does I'll pick it up when I'm feeling better

**Comment by FreddieAkeroyd** [comment]
Doing #6268 would be helpful as would avoid patching GALILMUL now and in the future

**Comment by FreddieAkeroyd** [comment]
code compiles, not sure if it works! Jenkins `new_galil_driver` job now restarted

**Comment by FreddieAkeroyd** [comment]
Waiting for office galil to be power cycled for initial testing. Currently investigating error message with profile moves when DBs are loaded and device not yet available, but it may not be anything to worry about

**Comment by FreddieAkeroyd** [comment]
Profile move errors disappear when `GalilCreateCSAxes("Galil")` added to local `galil.cmd`

**Comment by FreddieAkeroyd** [comment]
So connects to newly powered on galil OK, just need to compare PVs created from old and new galil drivers to see is anything is missing that we might be using. Then it can go for review.

**Comment by FreddieAkeroyd** [comment]
Comparison of PVS:

* DMC01:STOPE_STATUS has gone - did we use?
* EDEL_MON and EDEL_SP have gone, maybe replaced by ENC_TOLERANCE_MON and ENC_TOLERANCE_SP - check and maybe create an alias?
* HOMED_STATUS, HOMEVAL_MON, HOMEVAL_SP gone, see if covered by HOMING_ and HOMEALLOWED_
* STOPGO_CMD, STOPGO_MON gone - no equivalent? have STOP_ and STOPDELAY_ listed

Autosave

* many motor record bits missing, have ADEL, MDEL saved but missing rest. Probably merge error or not loading req. There are now separate templates depending on motor record version, so may affect req file naming
* EDEL_SP, HOMEVAL_SP, _able not saved
* also not saving parameters for CS axes 9-16, see if it should be. These were not created in old driver.

**Comment by John-Holt-Tessella** [comment]
Definitely use EDEL_MON and EDEL_SP  for stall detection on MERLIN I think with the jaws that are monitored via resistance and we needed to stop them redefining their position

**Comment by FreddieAkeroyd** [comment]
Testing:

* Do not merge the `Update to 3 4` PRs at this point  - they should wait until further testing on beamline
* Only PR to merge is ISISComputingGroup/EPICS-galil#53 to allow autosave files to be compatible between drivers for EDEL
* you should be able to run up new galil driver with existing autosave and settings\galil\galil01.cmd` etc  files and comunicate with e.g. R3 galil
* HOMEVAL_SP  is no more; home position is always zero. This may need to be changed on an instrument and an appropriate offset used instead.

**Comment by FreddieAkeroyd** [comment]
So now ready for review prior to beamline testing. After reviewing GMC code by eye it looked like some of the job speed variables were no longer being set by the new driver, so I have added then back in for now. Hopefully i have spotted anyt other missing bits, but I have been unable to test any homing code.   

At this stage:
- review code changes, changes from current vendor and 3_4 branch are less, really it is whether anything has missed being included. Unfortunately the diff between 3_4 and out current master are quite bug and it may be hard to tell, so may come down to lots of beamline testing.
- don't merge the 3_4 branch, we will test using the separate jenkins job
- only PR to consider is one related EDEL support   ISISComputingGroup/EPICS-galil#53

**Comment by DominicOram** [comment]
Jenkins build is failing but I believe I have fixed it for the next run. Otherwise looks about as good as I can tell without testing on a beamline
